core/src/test/java/com/linecorp/armeria/client/Http2ClientSettingsTest.java

```diff
@@ -0,0 +1,389 @@
+            bos.write(new byte[] { 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03 });
+            bos.write(new byte[1]); // Triggers the client to send a WINDOW_UPDATE frame for the connection.
+            bos.flush();
+
+            // Read an WINDOW_UPDATE frame and validate it.
+            assertConnectionWindowUpdateFrame(in);
+
+            // Send a DATA frame that indicates the end of stream id 03.
+            bos.write(new byte[] { 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x03 });
+            bos.flush();
+
+            future.join();
+        } finally {
+            Closeables.close(s, true);
+        }
+    }
+
+    @Test
+    public void maxFrameSize() throws Exception {
+
+        Socket s = null;
+        try (ServerSocket ss = new ServerSocket(0)) {
+            final int port = ss.getLocalPort();
+
+            final ClientFactory clientFactory = new ClientFactoryBuilder()
+                    .useHttp2Preface(true)
+                    .http2MaxFrameSize(DEFAULT_MAX_FRAME_SIZE * 2) // == 32 * 1024
+                    .build();
+
+            final HttpClient client = HttpClient.of(clientFactory, "http://127.0.0.1:" + port);
+            final CompletableFuture<AggregatedHttpMessage> future = client.get("/").aggregate();
+
+            s = ss.accept();
+            final InputStream in = s.getInputStream();
+            final BufferedOutputStream bos = new BufferedOutputStream(s.getOutputStream());
+
+            readBytes(in, connectionPrefaceBuf().capacity()); // Read the connection preface and discard it.
+
+            // Read a SETTINGS frame and validate it.
+            assertSettingsFrameOfMaxFrameSize(in);
+
+            sendEmptySettingsAndAckFrame(bos);
+
+            readBytes(in, 9); // Read a SETTINGS_ACK frame and discard it.
+            readHeadersFrame(in); // Read a HEADERS frame and discard it.
+
+            sendHeaderFrame(bos);
+
+            ///////////////////////////////////////
+            // Transmission of data gets started. //
+            ///////////////////////////////////////
+
+            // Send a DATA frame that indicates sending data as much as 0x8000 for stream id 03.
+            bos.write(new byte[] { 0x00, (byte) 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03 });
+            bos.write(EMPTY_DATA);
+            bos.write(EMPTY_DATA);
+            bos.flush();
+
+            readBytes(in, 13); // Read a WINDOW_UPDATE frame for connection and discard it.
+            readBytes(in, 13); // Read a WINDOW_UPDATE frame for stream id 03 and discard it.
+
+            // Send a DATA frame that exceed MAX_FRAME_SIZE by 1.
+            bos.write(new byte[] { 0x00, (byte) 0x80, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x03 });
+            bos.flush(); // Triggers the client to send a GOAWAY frame for the connection.
+
+            // The client send a GOAWAY frame and the server read it.
+            final ByteBuf buffer = readGoAwayFrame(in);
```

Comment:
If you are concerned about this log message being logged, you can edit `/build.gradle` to add this specific message for this specific test case to the exception list so it's not printed out to the console.